### ISSUE TITLE
Memory leak with static delimiters with reference at the root

### ISSUE THREAD

**Issue by marcalexiei** [ractivejs/ractive#3183]
## Description:
A memory leak is present after teardown of component which use static delimiters with `@.root` reference.

Issue occurs also with `@.parent`.

## Versions affected:
0.9.x

## Platforms affected:
All (tested on Chrome)

## Reproduction:
```js
Ractive.components.MyComponent = Ractive.extend({
  template: `
  <div>
    <h1>My component</h1>
    <p>[[@.root.get('description')]]</p>
  </div>
  `,
});

const r = window.r = new Ractive({
  el: '#main',
  template: `
  <h1>Title</h1>
  {{#if showComponent}}
  	<MyComponent/>
  {{/if}}
  `,
  data: {
    showComponent: true,
    description: 'Root description',
  }
});

setTimeout(() => {
  r.set({ showComponent: false });
}, 500);
```

https://jsfiddle.net/0b3dp3ch/

1. Using Chrome open fiddle and wait 1 second.
2. Perform a memory heap snapshot.

![screen shot 2018-01-25 at 12 02 45](https://user-images.githubusercontent.com/24919330/35385427-883e0f08-01c8-11e8-983c-87cdd49cffc6.png)

**Comment by evs-chris** [comment]
There's not really a good way to add a test for this, but I _think_ I have it resolved on edge now. If you could, verify that it is, and I'll add the fix to the queue for 0.9.12.

**Comment by marcalexiei** [comment]
The issue is fixed in edge build.
Thanks for the quick fix!
